source/fem-dem/gls_vans.cc

```diff
@@ -849,8 +885,21 @@ GLSVANSSolver<dim>::assemble_system_rhs()
                                     *this->cell_quadrature,
                                     *this->mapping);
 
+#if DEAL_II_VERSION_GTE(10, 0, 0)
+  max_particle_id = particle_handler.get_max_local_particle_index();
+#else
+  {
+    max_particle_id = 0;
+    for (const auto &particle : particle_handler)
+      {
+        max_particle_id = std::max(max_particle_id, particle.get_id());
+      }
+    max_particle_id += 1;
+  }
+#endif
+
   scratch_data.enable_particle_fluid_interactions(
-    particle_handler.n_global_max_particles_per_cell());
+    particle_handler.n_global_max_particles_per_cell(), max_particle_id);
```

Comment:
Is this needed for the global force storage structure? Shouldnt that global structure be used only in the copy local to global function?

Reply:
We were getting a segmentation fault in Debug mode. The issue was that the fluid_particle_force in the dem integration was accessing data that doesn't exist.So I had to initialize it to the same dimension as the the force in dem, as so they both use particle_id as index to store the calculated forces.

Reply:
Yeah but this means that every copy structure has a very very large data storage structure.
It would be better to create a smaller vector of the size of the number of particles in a cell. And also store the id of the particles in a cell in a similar vector. Then work with this small vector and at the end copy it into the larger vector using the smaller vector and the ID. Otherwise, you are creating something that will blow up when you will have a lot of particles per core. It's not a good design idea. We can talk about it if you want.

Reply:
done